packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -996,14 +998,16 @@ class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldCont
         // underscrolled, OR, going backward (fling down) and inner list is
         // scrolled past zero. We want to skip the pixels we don't need to grow
         // or shrink over.
+        double rawExtra = 0.0;
         if (velocity > 0.0) {
           // shrinking
-          extra = _outerPosition!.minScrollExtent - _outerPosition!.pixels;
+          rawExtra = _outerPosition!.minScrollExtent - _outerPosition!.pixels;
         } else if (velocity < 0.0) {
           // growing
-          extra = _outerPosition!.pixels - (_outerPosition!.maxScrollExtent - _outerPosition!.minScrollExtent);
+          rawExtra = _outerPosition!.pixels - (_outerPosition!.maxScrollExtent - _outerPosition!.minScrollExtent);
         }
-        assert(extra <= 0.0);
+        assert(rawExtra <= precisionErrorTolerance);
+        extra = math.min(0.0, rawExtra);
```

Comment:
I couldn't find a way to consitently reproduce this, see https://github.com/flutter/flutter/issues/79062. These changes should prevent situations like this

Reply:
I am not sure I follow, do you mean that this part of the change is untested?

Reply:
Yes, unfortunately

Reply:
If we cannot verify a change works as expected, I do not know that we should make that specific change.

Reply:
Ok, removed this